client/src/App.js

```diff
@@ -122,42 +148,39 @@ class App extends Component {
     this.renderLoggedInRoutes = this.renderLoggedInRoutes.bind(this);
   }
 
-  componentDidMount() {
+  async componentWillMount() {
+    var role;
+    var profile;
     this.setState({
       fetchingAuth: true
-    }, async () => {
-      try {
-        const result = await makeCall({}, '/isLoggedIn', 'get')
+    }) 
+    try {
+      const result = await makeCall({}, '/isLoggedIn', 'get')
+      this.setState({
+        fetchingAuth: false
+      });
+      if (result && !result.error) {
+        var jwtVal = document.cookie.replace(/(?:(?:^|.*;\s*)jwt\s*\=\s*([^;]*).*$)|^.*$/, "$1");
+        const parsedJWT = JSON.parse(atob(jwtVal.split('.')[1]));
+        role = parsedJWT.role;
+        profile = parsedJWT.details
+        this.setState({
+          role: role,
+          userDetails: profile,
+          loggedIn: true
+        })
+      } else {
         this.setState({
-          fetchingAuth: false
+          loggedIn: false,
+          fetchingAuth: false,
         });
-        if (result && !result.error) {
-          this.setState({
-            loggedIn: true
-          });
-        } else {
-          this.setState({
-            loggedIn: false,
-            fetchingAuth: false,
-          });
-        }
-      } catch (e) {
-          this.setState({
-            loggedIn: false,
-            fetchingAuth: false,
-          });
-          console.log("Error: App#componentDidMount", e)
-      }
-    })
-    const persistedState = JSON.parse(localStorage.getItem(App_LS));
```

Comment:
Persisted state was not being used due to quirks with setState in liftedPayload, and is now unnecessary

Reply:
Good riddance! Let's try to get rid of using localStorage all together!